contracts/modules/licensing/LicensingModuleV2.sol

```diff
@@ -0,0 +1,360 @@
+                originalIpIds[i],
+                licenseTemplate,
+                lcId
+            );
+            if (mlc.receiverCheckModule != address(0)) {
+                if (!IHookModule(mlc.receiverCheckModule).verify(derivativeIpOwner, mlc.receiverCheckData)) {
+                    revert Errors.LicensingModule__ReceiverCheckFailed(derivativeIpOwner);
+                }
+            }
+            (address royaltyPolicy, bytes memory royaltyData) = _payMintingFee(
+                originalIpIds[i],
+                licenseTemplate,
+                lcId,
+                1,
+                royaltyContext,
+                mlc
+            );
+            royaltyDatas[i] = royaltyData;
+
+            if (i == 0) {
+                commonRoyaltyPolicy = royaltyPolicy;
+            } else if (royaltyPolicy != commonRoyaltyPolicy) {
+                revert Errors.LicensingModule__IncompatibleRoyaltyPolicy(royaltyPolicy, commonRoyaltyPolicy);
+            }
+        }
+    }
+
+    function _payMintingFee(
+        address originalIpId,
+        address licenseTemplate,
+        uint256 licenseTermsId,
+        uint256 amount,
+        bytes calldata royaltyContext,
+        Licensing.MintingLicenseConfig memory mlc
+    ) private returns (address royaltyPolicy, bytes memory royaltyData) {
+        ILicenseTemplate lct = ILicenseTemplate(licenseTemplate);
+        uint256 mintingFee = 0;
+        address currencyToken = address(0);
+        (royaltyPolicy, royaltyData, mintingFee, currencyToken) = lct.getRoyaltyPolicy(licenseTermsId);
+
+        if (royaltyPolicy != address(0)) {
+            ROYALTY_MODULE.onLicenseMinting(originalIpId, royaltyPolicy, royaltyData, royaltyContext);
+            uint256 tmf = _getTotalMintingFee(mlc, originalIpId, licenseTemplate, licenseTermsId, mintingFee, amount);
+            // pay minting fee
+            if (tmf > 0) {
+                ROYALTY_MODULE.payLicenseMintingFee(originalIpId, msg.sender, royaltyPolicy, currencyToken, tmf);
+            }
+        }
+    }
+
+    function _getTotalMintingFee(
+        Licensing.MintingLicenseConfig memory mintingLicenseConfig,
+        address licensorIpId,
+        address licenseTemplate,
+        uint256 licenseTermsId,
+        uint256 mintingFeeSetByLicenseTerms,
+        uint256 amount
+    ) private view returns (uint256) {
+        if (!mintingLicenseConfig.isSet) return mintingFeeSetByLicenseTerms * amount;
+        if (mintingLicenseConfig.mintingFeeModule == address(0)) return mintingLicenseConfig.mintingFee * amount;
+        return
+            IMintingFeeModule(mintingLicenseConfig.mintingFeeModule).getTotalMintingFee(
+                licensorIpId,
+                licenseTemplate,
+                licenseTermsId,
+                amount
+            );
+    }
+
+    /// @dev Verifies if the IP is disputed
+    function _verifyIpNotDisputed(address ipId) private view {
```

Comment:
We may not need to a view function that calls another existing view function to return the same result.

Also, for consistency we can remove the _verifyIpNotDisputed and just use DISPUTE_MODULE.isIpTagged(ipId) like in royalty contracts.

Reply:
Agreed. the function mainly for reduce duplicate code.